keras_cv/models/convnext.py

```diff
@@ -0,0 +1,539 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ConvNeXt models for Keras.
+
+References:
+- [A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545)
+  (CVPR 2022)
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+from keras_cv.layers.regularization import StochasticDepth
+from keras_cv.models import utils
+
+MODEL_CONFIGS = {
+    "tiny": {
+        "depths": [3, 3, 9, 3],
+        "projection_dims": [96, 192, 384, 768],
+        "default_size": 224,
+    },
+    "small": {
+        "depths": [3, 3, 27, 3],
+        "projection_dims": [96, 192, 384, 768],
+        "default_size": 224,
+    },
+    "base": {
+        "depths": [3, 3, 27, 3],
+        "projection_dims": [128, 256, 512, 1024],
+        "default_size": 224,
+    },
+    "large": {
+        "depths": [3, 3, 27, 3],
+        "projection_dims": [192, 384, 768, 1536],
+        "default_size": 224,
+    },
+    "xlarge": {
+        "depths": [3, 3, 27, 3],
+        "projection_dims": [256, 512, 1024, 2048],
+        "default_size": 224,
+    },
+}
+
+BASE_DOCSTRING = """Instantiates the {name} architecture.
+    - [A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545)
+    (CVPR 2022)
+
+    This function returns a Keras {name} model.
+
+    Args:
+        include_rescaling: whether or not to Rescale the inputs.If set to True,
+            inputs will be passed through a `Rescaling(1/255.0)` layer.
+        include_top: whether to include the fully-connected layer at the top of the
+            network.  If provided, classes must be provided.
+        depths: an iterable containing depths for each individual stages.
+        projection_dims: An iterable containing output number of channels of
+            each individual stages.
+        drop_path_rate: stochastic depth probability, if 0.0, then stochastic
+            depth won't be used.
+        layer_scale_init_value: layer scale coefficient, if 0.0, layer scaling
+            won't be used.
+        weights:
```

Comment:
Not sure what do I put here. `weights.py` doesn't have any weights for ConvNeXt.

Reply:
For now, this will only support loading weights from a directory (no pre-trained weights).

If you'd like to train some weights, feel free to do so using basic_training.py or by contributing your own training script.